typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -155,7 +175,13 @@ export class MultiProvider<MetaExt = {}> extends ChainMetadataManager<MetaExt> {
     if (signer.provider) return signer;
     // Auto-connect the signer for convenience
     const provider = this.tryGetProvider(chainName);
-    return provider ? signer.connect(provider) : signer;
+    if (!provider) return signer;
+
+    // Handle ZKSync provider separately
+    if (signer instanceof ZKSyncWallet) {
```

Comment:
Using `instanceof` this way has pitfalls and is generally a code smell. Is there a tangible property on the signer you can check instead?

Reply:
Your comment on removing ZKSyncWallet form ChainMap helped me resolve this. No longer need to check for instance of